src/GeneratedHydrator/CodeGenerator/Visitor/HydratorMethodsVisitor.php

```diff
@@ -22,6 +22,17 @@
 class HydratorMethodsVisitor extends NodeVisitorAbstract
 {
     /**
+     * When this option is passed, only the properties in the given array are
+     * hydrated and extracted.
+     */
+    const OPTION_ALLOWED_PROPERTIES = 'allowedProperties';
```

Comment:
Can probably inject it as a separate array, instead of injecting all options (makes it easier to deal with)